### DISCUSSION TITLE
MarkovJunior Web Version

### DISCUSSION THREAD

**Discussion by Yuu6883** [mxgmn/MarkovJunior#41] [Ideas]
I made a web version of this: https://github.com/Yuu6883/MarkovJuniorWeb
It's finished now with all the nodes correctly implemented 🎉
The code does run about 2.5x slower than the C# version, but it's quite possible to be optimized with WebAssembly.

Here's a screenshot of the page:
![image](https://user-images.githubusercontent.com/38842891/176581431-c00c2e0e-8b25-4e18-afee-cf98579635c3.png)

**Comment by kaya3** [comment]
I don't know if you're aware, but there are some other people working on porting MarkovJunior to TypeScript; you might want to take a look at that project to avoid duplicated effort.

GitHub repo for their project: https://github.com/aseaday/MarkovJunior.js

**Reply by Yuu6883** [reply]
Oh that's nice to know, I should have searched before I started. My progress is more or less the same with that repo.

**Comment by Yuu6883** [comment]
Looking good now
![image](https://user-images.githubusercontent.com/38842891/175770381-69375b24-c53d-4f37-9138-a57c62faa61e.png)

**Comment by mxgmn** [comment]
Cool! I linked it from the README.

**Comment by lalawuu** [comment]
Better than my progress. I am stuck with the debug and performance issue.
